src/transform/FooterReadMore.js

```diff
@@ -0,0 +1,336 @@
+ * @param {!string} closer
+ * @return {!string}
+ */
+const safelyRemoveEnclosures = (string, opener, closer) => {
+  const enclosureRegex = new RegExp(`\\s?[${opener}][^${opener}${closer}]+[${closer}]`, 'g')
+  let counter = 0
+  const safeMaxTries = 30
+  let stringToClean = string
+  let previousString = ''
+  do {
+    previousString = stringToClean
+    stringToClean = stringToClean.replace(enclosureRegex, '')
+    counter++
+  } while (previousString !== stringToClean && counter < safeMaxTries)
+  return stringToClean
+}
+
+/**
+ * Removes '(...)' and '/.../' parenthetical enclosures from string. 
+ * @param {!string} string
+ * @return {!string}
+ */
+const cleanExtract = string => {
+  let stringToClean = string
+  stringToClean = safelyRemoveEnclosures(stringToClean, '(', ')')
+  stringToClean = safelyRemoveEnclosures(stringToClean, '/', '/')
+  return stringToClean
+}
+
+/**
+ * Read more page model.
+ */
+class ReadMorePage {
+  /**
+   * ReadMorePage constructor.
+   * @param {!string} title
+   * @param {?string} thumbnail
+   * @param {?object} terms
+   * @param {?string} extract
+   * @return {void}
+   */
+  constructor(title, thumbnail, terms, extract) {
+    this.title = title
+    this.thumbnail = thumbnail
+    this.terms = terms
+    this.extract = extract
+  }
+}
+
+/**
+ * Makes document fragment for a read more page.
+ * @param {!ReadMorePage} readMorePage
+ * @param {!number} index
+ * @param {SaveButtonClickHandler} saveButtonClickHandler
+ * @param {!Document} document
+ * @return {!DocumentFragment}
+ */  
+const documentFragmentForReadMorePage = (readMorePage, index, saveButtonClickHandler, document) => {
+  const outerAnchorContainer = document.createElement('a')
+  outerAnchorContainer.id = index
+  outerAnchorContainer.className = 'pagelib_footer_readmore_page'
+
+  const hasImage = readMorePage.thumbnail && readMorePage.thumbnail.source
+  if (hasImage) {
+    const image = document.createElement('div')
+    image.style.backgroundImage = `url(${readMorePage.thumbnail.source})`
+    image.classList.add('pagelib_footer_readmore_page_image')
+    outerAnchorContainer.appendChild(image)
+  }
+
+  const innerDivContainer = document.createElement('div')
+  innerDivContainer.classList.add('pagelib_footer_readmore_page_container')
+  outerAnchorContainer.appendChild(innerDivContainer)
+  outerAnchorContainer.href = `/wiki/${encodeURI(readMorePage.title)}`
+
+  if (readMorePage.title) {
+    const title = document.createElement('div')
+    title.id = index
+    title.className = 'pagelib_footer_readmore_page_title'
+    const displayTitle = readMorePage.title.replace(/_/g, ' ')
+    title.innerHTML = displayTitle
+    outerAnchorContainer.title = displayTitle
+    innerDivContainer.appendChild(title)
+  }
+
+  let description
+  if (readMorePage.terms) {
+    description = readMorePage.terms.description[0]
+  }
+  if ((description === undefined || description.length < 10) && readMorePage.extract) {
```

Comment:
This could be concisely written as `(description || '').length < 10` or `(!description || description.length < 10)`.